### ISSUE TITLE
dsl: create a written tutorial for developers (other teams) to use the dsl

### ISSUE THREAD

**Issue by wzeng0** [uchicago-cs/chiventure#1256]
team members: Hira, Jay, Andi

This task is to create a detailed User Guide for Other Teams/Game Devlopers to use the DSL format. This can be useful for testing, since instead of having to working through the complex syntax of the WDL format currently, they will be able to utilize the more user-friendly DSL which will save time/energy in the long run. The following is a list of items that we are hoping to add in the guide (which will be in the format of a Wiki page).

- Create a README/User Guide for a Game Developer using a DSL format:
-- What are the potential elements they can add to the game?
-- A detailed description of the format (tabs, headers, etc).
-- A list of elements (e.g. sound) that the game developer might have wanted to add but are not in development yet.
-- Nested events/complex story structures
-- Basically, updating the (User stories) Wiki Page to be more detailed for ease of User (specific examples related to game development)

**Comment by awliu2** [comment]
Hi guys, made a document where we can draft out the tutorial. I think we should make a wiki page with step by step screenshots, and we can also update the readme as needed.

https://docs.google.com/document/d/1d8QHaXYi4Z3WkqhAn1w6zxdnfZLoR79SdqSS94GcBWg/edit?usp=sharing

the contents of this google doc have largely been included either below, or on the wiki, so I don' think it would make sense to copy paste it here (also its like 4 pages of largely non-connected sections.)

**Comment by jhsivadas** [comment]
Hi guys, I made a wikipage for the DSL tutorial.

https://github.com/uchicago-cs/chiventure/wiki/DSL-~-DSL-Tutorial

**Comment by baigh1** [comment]
Updated the Wiki Page to include a short example (also listed in the google doc).

**Comment by baigh1** [comment]
Added a section giving a brief overview of what is supported in the DSL.

**Comment by baigh1** [comment]
Added a section for defining the syntax + extra notes on the use of variables in the DSL.

**Comment by awliu2** [comment]
We've finished updating the wiki to provide much more specific, user-focused information. We described the possible applications of DSL, how to run the parser and set up the environment, and provided many examples of various use cases. We also included syntax and format guidelines.

**Comment by jacknugent1529** [comment]
Issue Score: Excellent

Comments: Good job! Some of the updates could be a bit more detailed, but the frequency of updates is fantastic.
